app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt

```diff
@@ -65,6 +65,11 @@ class UserPreferences(context: Context) : SharedPreferenceStore(
 		 */
 		var nextUpTimeout = Preference.int("next_up_timeout", 1000 * 7)
 
+		/**
+		 * Display the next up item's thumbnail on the next up screen
+		 */
+		var nextUpThumbnailEnabled = Preference.boolean("next_up_thumbnail_enabled", true)
```

Comment:
I'm still in favor of having a generic option for hiding spoilers to be used for these types of features instead of individual options.

Reply:
I too think that's a better approach.

Reply:
Regarding this specific option (hiding thumbnails on the next up screen), I don't think it should be included in a hide spoilers feature. Hiding spoilers should apply only to unwatched items, whereas this option hides the thumbnail whether the episode is watched or not. Since nothing appears on the next up screen until everything is loaded, loading the thumbnail makes the episode title and logo take longer to appear. In my experience, it sometimes takes 5+ seconds for the thumbnail to load, meaning the next up screen is blank for almost the entire duration (or the entire duration) the screen is set to be shown before starting the episode. Without having to load the thumbnail, the title and logo appear instantly since the logo image is cached after watching one episode of the show. Plus, in my opinion, the layout looks better without the thumbnail. 

Regarding options to hide spoilers, I'm not sure I understand why it would be better to group the options together. There could be spoilers in thumbnails, descriptions, episode titles, cast information, and video chapters. These things make up most of the metadata shown in the app, and I don't think it's safe to assume that if someone wants one of these things hidden they want all of them hidden. 

What if someone wants thumbnails hidden, but they like having episode titles, movie descriptions, and cast information? Their only option would be to hide everything.

Episode descriptions contain spoilers more often than movie descriptions or series descriptions; someone might want episode descriptions to be hidden since they're already watching the show but still want movie descriptions or series descriptions to be shown when they're deciding on something new to watch. 

The only reason I can think of to group the options into one generic option would be to prevent clutter in the settings menu, which is why I think it would be good to include them in a submenu like how the 'Developer options' are implemented—this way the options would only take up one slot in the main settings menu.

Reply:
What we _could_ do is to change the "enable next up screen" option to an enum with these options:
- Disabled
- Enabled, minimal (no thumb)
- Enabled, extended (yes thumb)

That way the user can chose how it looks without adding an additional preference.